aptos-move/e2e-tests/src/executor.rs

```diff
@@ -377,6 +381,64 @@ impl<O: OutputLogger> FakeExecutorImpl<O> {
         self
     }
 
+    /// Enable code coverage collection for this executor. This is thread safe,
+    /// but all threads should use the same `file_base_name` to avoid logical confusion. If
+    /// multiple threads with the same file generate tracing info concurrently, it will be
+    /// accumulated as expected. See `save_code_coverage` how to persist coverage info
+    /// and how to analyze it.
+    pub fn enable_code_coverage(&self, file_base_name: &str) {
+        // Ensure debugging is enabled in the VM
+        aptos_vm_environment::prod_configs::set_debugging_enabled(true);
+        // Enable execution tracing.
+        let trace_name = format!("{}.mvtr", file_base_name);
+        tracing::enable_tracing(Some(&trace_name));
+    }
+
+    /// Saves code coverage information collected so far into a coverage file. If `append`
+    /// is true, an existing coverage file is extended. The file is located
+    /// at `"{file_base_name}.mvcov"`. The information collected so far is cleared.
+    ///
+    /// This function is idempotent, that is, if called multiple times in sequence in append
+    /// mode, it only adds information not yet collected so far to the coverage file.
+    ///
+    /// A typical way how this is used from code is as follows:
+    ///
+    /// ```ignore
+    ///   executor.enable_code_coverage(coverage_file);
+    ///   for test in tests {
+    ///     test.run();
+    ///     executor.save_code_coverage(coverage_file);
+    ///   }
+    /// ```
+    ///
+    /// The output in `{file_base_name}.mvcov` can be fed in to Aptos CLI for analysis of coverage
+    /// for a given package:
+    ///
+    /// ```ignore
+    ///   package_dir> aptos move \
+    ///      coverage summary --extra-coverage {file_base_name}.mvcov
+    ///   package_dir> aptos move \
+    ///      coverage source --extra-coverage {file_base_name}.mvcov --module my_module
+    /// ```
+    ///
+    /// The coverage in the file will be added to any coverage produced by
+    /// `aptos move test --coverage`, such that both unit tests and e2e tests count
+    /// for overall coverage.
+    ///
+    pub fn save_code_coverage(&self, file_base_name: &str, append: bool) -> anyhow::Result<()> {
```

Comment:
Is there ever a reason for having `append=false`, especially in the presence of threads? If not, we can probably get rid of it, avoiding incorrect callers.

Reply:
Good point, removed.